modules/pytket-aqt/tests/backend_test.py

```diff
@@ -168,6 +168,10 @@ def test_shots_bits_edgecases(n_shots, n_bits) -> None:
     assert res.get_counts() == correct_counts
 
     # Direct
-    assert np.array_equal(aqt_backend.get_shots(c, n_shots), correct_shots)
-    assert aqt_backend.get_shots(c, n_shots).shape == correct_shape
-    assert aqt_backend.get_counts(c, n_shots) == correct_counts
+    assert np.array_equal(
+        aqt_backend.run_circuit(c, n_shots=n_shots).get_shots(), correct_shots
```

Comment:
In this and similar tests elsewhere, can we just get the `BackendResult` once and call the methods on it? Otherwise I think we are processing the circuit multiple times?